### PR TITLE
update armature

### PR DESCRIPTION
**Pull request by 2youyou2**
1.fixed set displayName when addDisplay
2.change play several movement method
3.support read tweenRotate property from dragonBone
4.fixed memory leak

### CODE REVIEW

File: extensions/CocoStudio/Armature/animation/CCArmatureAnimation.cpp [hunk 1]
```diff
@@ -260,31 +261,31 @@ void CCArmatureAnimation::playByIndex(int animationIndex, int durationTo, int du
...
 
     updateMovementList();
 }
 
-void CCArmatureAnimation::playByIndex(bool loop, const int *movementIndexes, int movementNumber)
+void CCArmatureAnimation::playByIndex(const std::vector<int>& movementIndexes, int durationTo, bool loop)
 {
     m_sMovementList.clear();
     m_bMovementListLoop = loop;
+    m_iMovementListDurationTo =durationTo;
```
Comment (dumganhar) [inline extensions/CocoStudio/Armature/animation/CCArmatureAnimation.cpp]: Please add a space after `=`.

File: extensions/CocoStudio/Armature/animation/CCArmatureAnimation.cpp [hunk 2]
```diff
@@ -293,6 +294,28 @@ void CCArmatureAnimation::playByIndex(bool loop, const int *movementIndexes, int
...
+    std::vector<std::string> names;
+    
+    CCObject *object = NULL;
+    CCARRAY_FOREACH(movementNames, object)
+    {
+        names.push_back(static_cast<CCString*>(object)->getCString());
+    }
+    play(names, durationTo, loop);
+}
+void CCArmatureAnimation::playWithIndexArray(CCArray *movementIndexes, int durationTo, bool loop)
```
Comment (dumganhar) [inline extensions/CocoStudio/Armature/animation/CCArmatureAnimation.cpp:309]: Needs an empty line here.

File: extensions/CocoStudio/Armature/animation/CCArmatureAnimation.h [hunk 3]
```diff
@@ -149,12 +149,16 @@ class  CCArmatureAnimation : public CCProcessBase
     /**
      * Play several animation by names
      */
-    virtual void play(bool loop, const std::string *movementNames, int movementNumber);
-
-    /**
-     * Play several animation by index
+    virtual void play(const std::vector<std::string>& movementNames, int durationTo = -1, bool loop = true);
+	/**
```
Comment (dumganhar) [inline extensions/CocoStudio/Armature/animation/CCArmatureAnimation.h]: Wrong indention.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp [hunk 4]
```diff
@@ -1363,13 +1363,14 @@ void TestPlaySeveralMovement::onEnter()
...
-    armature = cocos2d::extension::CCArmature::create("Cowboy");
-    armature->getAnimation()->play(true, names, 3);
-    //armature->getAnimation()->playByIndex(true, indexes, 3);
+    // 	std::string name[] = {"Walk", "FireMax", "Fire"};
+    // 	std::vector<std::string> names(name, name+3);
+    int index[] = {0, 1, 2};
+    std::vector<int> indexes(index, index+3);
+
+    CCArmature *armature = CCArmature::create("Cowboy");
+    //armature->getAnimation()->play(names);
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp]: Please remove unneeded codes.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp [hunk 5]
```diff
@@ -1363,13 +1363,14 @@ void TestPlaySeveralMovement::onEnter()
...
 
-    std::string names[] = {"Walk", "FireMax", "Fire"};
-    //int indexes[] = {0, 1, 2};
-
-    cocos2d::extension::CCArmature *armature = NULL;
-    armature = cocos2d::extension::CCArmature::create("Cowboy");
-    armature->getAnimation()->play(true, names, 3);
-    //armature->getAnimation()->playByIndex(true, indexes, 3);
+    // 	std::string name[] = {"Walk", "FireMax", "Fire"};
+    // 	std::vector<std::string> names(name, name+3);
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp]: Please remove unneeded codes.

Comment (2youyou2) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp]: These code is to teach users how to use 

std::string name[] = {"Walk", "FireMax", "Fire"};
std::vectorstd::string names(name, name+3);

as a param

**Comment by dumganhar** [comment]
Merged, thanks.

**Pull request merged**